### PR TITLE
Responsive cleanup / List Items

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

if merged this PR adds a breakpoint to assist in the list items in smaller viewports.

### :hammer_and_wrench: Detailed description

The `tokens-list`, `font-helpers-list`, and `icons-list` all break at around less that 500px. I added a new `x-small` breakpoint to account for the smaller viewport (though now the mixins are starting to become a little non-semantic).

All of these pages use relatively the same structure so I mirrored the stacking in all three.

### :camera_flash: Screenshots

Tokens before
tokens-before

Tokens after
tokens-after

Font helpers before
typography-before

Font helpers after
typography-after

Icons before
icons-before

Icons after
icons-after

### :link: External links

No jira ticket

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by heatherlarsen — APPROVED** [review]
I'll let @didoo review the code, but this is a great change visually. Thanks, @jorytindall!

**Comment by MelSumner** [comment]
@jorytindall this feels a little messy to me; if we do ship this now, would you create a backlog ticket to revisit and potentially tidy up things later? Thanks for considering!

**Review by didoo — COMMENTED** [review]
@jorytindall apart from the comment about avoiding a new breaking point value (and mixin) I am not too worried about the code, for me is ok. one thing I would suggest though is to try to use a mobile-first approach, which is what we have tended to do in other parts of the codebase. Essentially, have the basic behaviour (flex with vertical stacking) by default, and create a media-query that acts above a certain size for the grid-based layout. 

(if you need help in testing this alternative approach, just ping me)

**Comment by jorytindall** [comment]
@didoo I updated the breakpoints in each of these components to be mobile first, I think it works. The one thing that tripped me up is the `display: contents` property which if I understand inherits the grid specs of the larger container? Might be good to merge now, but I'll brush up on the spec.

File: website/app/styles/doc-components/font-helpers-list.scss [hunk 1]
```diff
@@ -1,36 +1,58 @@
...
+    border-radius: 3px 0 0 3px;
+  }
 }
 
 .doc-font-helpers-list__content {
   padding: 16px;
-  border: 1px solid var(--doc-color-gray-500);
-  border-left: none;
-  border-radius: 0 3px 0 3px;
+  border: solid var(--doc-color-gray-500);
```
Comment (didoo) [inline website/app/styles/doc-components/font-helpers-list.scss:48]: Nitpick: this declaration in interpreted by the browser as
```
border: initial solid var(--doc-color-gray-500);
```
(where the `initial` value in the browser is usually `1px`)

What you intend to do here, I think, is:
```
border-style: solid;
border-color: var(--doc-color-gray-500);
```

**Review by didoo — APPROVED** [review]
Left only a minor comment. All good to go.

**Comment by didoo** [comment]
> The one thing that tripped me up is the `display: contents` property which if I understand inherits the grid specs of the larger container? Might be good to merge now, but I'll brush up on the spec.

@jorytindall to make it simple: the `display: contents` property makes the container a "ghost"  in term of layout, it's like a "pass-through" in terms of CSS layout property. Reference: https://developer.mozilla.org/en-US/docs/Web/CSS/display#box

**Pull request merged**